hypothesis-python/src/hypothesis/extra/cli.py

```diff
@@ -84,27 +84,60 @@ def obj_name(s: str) -> object:
             return importlib.import_module(s)
         except ImportError:
             pass
+        classname = None
         if "." not in s:
             modulename, module, funcname = "builtins", builtins, s
         else:
             modulename, funcname = s.rsplit(".", 1)
             try:
                 module = importlib.import_module(modulename)
             except ImportError as err:
+                try:
+                    modulename, classname = modulename.rsplit(".", 1)
+                    module = importlib.import_module(modulename)
+                except ImportError as err:
+                    raise click.UsageError(
+                        f"Failed to import the {modulename} module for introspection.  "
+                        "Check spelling and your Python import path, or use the Python API?"
+                    ) from err
+        if classname is None:
+            try:
+                return getattr(module, funcname)
+            except AttributeError as err:
+                public_names = [
+                    name for name in vars(module) if not name.startswith("_")
+                ]
+                matches = get_close_matches(funcname, public_names)
                 raise click.UsageError(
-                    f"Failed to import the {modulename} module for introspection.  "
-                    "Check spelling and your Python import path, or use the Python API?"
+                    f"Found the {modulename!r} module, but it doesn't have a "
+                    f"{funcname!r} attribute."
+                    + (f"  Closest matches: {matches!r}" if matches else "")
+                ) from err
+        else:
+            try:
+                func_class = getattr(module, classname)
+            except AttributeError as err:
+                public_names = [
+                    name for name in vars(module) if not name.startswith("_")
+                ]
+                matches = get_close_matches(classname, public_names)
+                raise click.UsageError(
+                    f"Found the {modulename!r} module, but it doesn't have a "
+                    f"{classname!r} class."
+                    + (f"  Closest matches: {matches!r}" if matches else "")
+                ) from err
+            try:
+                return getattr(func_class, funcname)
+            except AttributeError as err:
+                public_names = [
+                    name for name in vars(func_class) if not name.startswith("_")
+                ]
+                matches = get_close_matches(funcname, public_names)
+                raise click.UsageError(
+                    f"Found the {modulename!r} module and {classname!r} class, "
+                    f"but it doesn't have a {funcname!r} attribute."
```

Comment:
Here I think it *is* worth distinguishing: we should check that `inspect.isclass(func_class)` before claiming that it's a class; at this stage all we know is that we imported something and then accessed an attribute!

Reply:
added check, if not class we say it's an attribute